apps/prairielearn/src/middlewares/authzWorkspace.js

```diff
@@ -44,8 +42,11 @@ module.exports = asyncHandler(async (req, res, next) => {
     await authzCourseOrInstance(req, res);
 
     if (isEnterprise()) {
-      const checkPlanGrants = promisify(require('../ee/middlewares/checkPlanGrants').default);
-      await checkPlanGrants(req, res);
+      const hasPlanGrants = await checkPlanGrantsForLocals(res.locals);
+      if (!hasPlanGrants) {
+        res.redirect(`/pl/course_instance/${res.locals.course_instance.id}/upgrade`);
+        return;
+      }
```

Comment:
This was actually subtly broken before. The middleware in use before will redirect if the user doesn't have the necessary plan grants, but we had no way of knowing if it redirected, so we'd just continue one processing the request and would probably eventually fail with "you can't set headers again" when we tried to respond later on.